### DISCUSSION TITLE
Inline text editor

### DISCUSSION THREAD

**Discussion by JankovicsSandor** [AvaloniaUI/Avalonia#6458] [Q&A] [answered]
Hi everyone!

I would like to implement an inline text editor for my application, such that when the user clicks on a text it transforms to an editable text and when escapes from the box it turns back to a normal label. I was thinking about doing someting with the combination of textbox focus and lostFocus events and replace the controls somehow. Is there any such functionality? or it needs to be done manually

Something like this:
![image](https://user-images.githubusercontent.com/25934939/130491278-8bc623eb-3863-4630-9d52-237087ef372e.png)
After click
![image](https://user-images.githubusercontent.com/25934939/130491313-6c1670cc-9084-40dd-8e3e-af7338b991fc.png)

Every idea helps.
Thanks

**Comment by wieslawsoltes** [comment, marked as answer]
I have made something similar using Behaviors 

https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Views/Containers/ProjectContainerView.axaml#L77-L144

Here you init the editing (key press or double-tapped):
https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Views/Containers/ProjectContainerView.axaml#L81-L82

https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Behaviors/ShowOnDoubleTappedBehavior.cs#L10

https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Behaviors/ShowOnKeyDownTappedBehavior.cs#L10

Here you hide editor (key pressed or lost focus):
https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Views/Containers/ProjectContainerView.axaml#L93-L96

https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Behaviors/HideOnKeyPressedBehavior.cs#L10

https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Behaviors/HideOnLostFocusBehavior.cs#L10

https://github.com/wieslawsoltes/Core2D/blob/deb7617f1b883f2acc14d1ac125f3de921595573/src/Core2D/Behaviors/SelectAllOnGotFocusBehavior.cs#L9

**Reply by JankovicsSandor** [reply]
Thanks it solved the problem :)

**Comment by wieslawsoltes** [comment]
I have added behaviors from my example to NuGet package https://www.nuget.org/packages/Avalonia.Xaml.Interactions/

* Sample implementation for ListBox

https://github.com/wieslawsoltes/AvaloniaBehaviors/blob/master/samples/BehaviorsTestApplication/Views/Pages/EditableListBoxView.axaml

* Sample implementation for TreeView

https://github.com/wieslawsoltes/AvaloniaBehaviors/blob/master/samples/BehaviorsTestApplication/Views/Pages/EditableTreeViewView.axaml

* Sample implementation for Item template (used in both ListBox and TreeView)

https://github.com/wieslawsoltes/AvaloniaBehaviors/blob/master/samples/BehaviorsTestApplication/Views/ItemView.axaml
